pyccel/codegen/printing/fcode.py

```diff
@@ -2684,9 +2684,16 @@ def _print_NumpySign(self, expr):
             numpy_sign is an interface which calls the proper function depending on the data type of x
 
         """
-        func = PyccelFunctionDef('numpy_sign', NumpySign)
-        self._additional_imports.add(Import('numpy_f90', AsName(func, 'numpy_sign')))
-        return f'numpy_sign({self._print(expr.args[0])})'
+        arg = expr.args[0]
+        arg_code = self._print(arg)
+        if isinstance(expr.dtype, NativeComplex):
+            func = PyccelFunctionDef('numpy_sign', NumpySign)
+            self._additional_imports.add(Import('numpy_f90', AsName(func, 'numpy_sign')))
+            return f'numpy_sign({arg_code})'
+        else:
+            cast_func = DtypePrecisionToCastFunction[expr.dtype.name][expr.precision]
+            abs_result = self._print(cast_func(PythonBool(arg)))
```

Comment:
What does `PythonBool` achieve here ?

Reply:
It's a cheat's way of checking if `arg!=0` it lets me avoid duplicating the associated code. I'll add a comment

Reply:
ce2e6f439